fzf-marks.plugin.bash

```diff
@@ -22,12 +22,25 @@
 
 command -v fzf >/dev/null 2>&1 || return
 
+PROGRAM_NAME="${PROGRAM_NAME:-fzf-marks}"
+
+CONFIG_DIR="${XDG_CONFIG_HOME:-$HOME/.config}/$PROGRAM_NAME" #currently unused, but in the future can be used to load (and store?) an .env file from to more easily switch between multiple configuration
+DATA_DIR="${XDG_DATA_HOME:-$HOME/.local/share}/$PROGRAM_NAME"
+LEGACY_FZF_MARKS_FILE="${HOME}/.fzf-marks"
```

Comment:
I suggest adding the prefix `_fzm_` to these private variable names. `fzf-marks` is not a separate shell program but a plugin to be sourced in interactive sessions, which means that the variable namespace is shared with other plugins.

Reply:
good point! I'll make the relevant changes. :)

Reply:
`PROGRAM_NAME` is also a variable that could conflict with other plugins. Also, `_fzm_PLUGIN_NAME` might be better as `fzf-marks` is not a program but just a plugin?

But, in the first place, do we need to make the plugin name `fzf-marks` configurable? Is there a use case?

Reply:
would make it easier for users to override the name of the sub-directories; personally I'll probably set it to fzf/marks/, to keep settings and data files of other fzf files grouped together.
Though I guess calling it `_fzm_xdg_subdir` might be a more appropiate name then?

Reply:
Oh, I see! This seems actually not a private variable. Then, how about `FZF_MARKS_XDG_SUBDIR`? The variables for the user settings are prefixed by `FZF_MARKS_`.